File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        
        int n     = height.length;
        int left  = 0 ;
        int right = n-1;
        int result = Integer.MIN_VALUE;
        while( left<right )
        {
            int Width   = right - left;
            int Length    = Math.min(height[left] , height[right]);
            int currArea = Length * Width;
            result       = Math.max(result , currArea );
            if( height[right] > height[left] )
            {
                left++;
            }else{
                right--;
            }
        }
        return result;
    }
}

File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int maxiArea = 0;
        int n        = height.size();
        int left    = 0;
        int right    = n-1;
        if( height.size() == 1 || height.size() == 0)return 0;
        while(left < right){
            int area = min(height[left] , height[right])*(right-left);
            maxiArea = max(maxiArea , area);
            
            if( height[left] >= height[right]){
                right--;
            }else{
                left++;
            }
        }

        return maxiArea;
    }
};